src/main/java/org/jenkinsci/plugins/authorizeproject/strategy/SpecificUsersAuthorizationStrategy.java

```diff
@@ -260,7 +262,11 @@ protected SpecificUsersAuthorizationStrategy newInstanceWithoutAuthentication(
             if (StringUtils.isBlank(userid)) {
                 throw new FormException("userid must be specified", "userid");
             }
-            
+            // TODO use Jenkins.getInstance().getSecurityRealm().getUserIdStrategy().equals(userid, ACL.SYSTEM.getPrincipal().toString())) once Jenkins 1.566+
+            if (userid.equals(ACL.SYSTEM.getPrincipal())) {
+                throw new FormException(Messages.SpecificUsersAuthorizationStrategy_userid_notSystem(), "userid");
+            }
+

```

Comment:
Correct, but we depend on the implementation of the security realm here... And there may be other code paths that make invalid assumptions and end up with the ACL.SYSTEM authentication cached in a reverse lookup map.

The result of all that is:

1. A correctly written Security Realm should never let you use SYSTEM with the specific user strategy (so we need to both alert the user that it is not supported and provide the admin with a way to do it)
2. An incorrectly written Security Realm might permit it, so we need to block any attempts to probe code paths that might expose such an issue.
3. I probably should do some filing of SECURITY issues against the code paths I have in my head and do not want to write in a public forum (assuming my gut feeling is correct)